### PR TITLE
Remove SetConsoleCtrlHandler from Unix PAL

### PR DESCRIPTION
**Pull request by stephentoub**
Deleted SIGINT and SIGQUIT-related signal handling code, which means we no longer need to spin up a worker thread for all coreclr apps.

Make the debugger's SetConsoleCtrlHandler usage Windows-only (from offline conversation it sounded like we might choose to delete it entirely subsequently, but for now I've left it).

Removed all of the code related to SetConsoleCtrlHandler from the PAL.

Deleted all tests related to SetConsoleCtrlHandler from the palsuite.

cc: @jkotas, @mikem8361, @janvorli, @kouvel 
(This should not be merged until https://github.com/dotnet/corefx/pull/4313 is merged in corefx.)

### CODE REVIEW

File: src/vm/ceemain.cpp [hunk 1]
```diff
@@ -832,7 +834,9 @@ void EEStartupHelper(COINITIEE fFlags)
         DisableGlobalAllocStore();
 #endif //_DEBUG
 
+#ifndef PLATFORM_UNIX
```
Comment (jkotas) [inline src/vm/ceemain.cpp]: CoreCLR typically uses `#ifndef FEATURE_PAL` in these type of situations.

**Comment by jkotas** [comment]
LGTM. Thanks!

**Comment by janvorli** [comment]
LGTM! It is nice to see such cleanup.

**Comment by adityamandaleeka** [comment]
LGTM. PAL keeps getting smaller :smile:

File: src/vm/ceemain.cpp (hunk 1 above)
Comment (stephentoub) [inline src/vm/ceemain.cpp]: Thanks. Fixed.

**Comment by stephentoub** [comment]
Thanks, guys.

**Comment by stephentoub** [comment]
@dotnet-bot test this please ("Failed to fetch from Git")

**Pull request merged**